OSDOCS-9565: configuring AWS Outposts postinstallation (#71263)
openshift/openshift-docs: modules/aws-outposts-prereq-storage-csi.adoc

Author: To-do: remove before final review/merge

// Module included in the following assemblies:
//
// * post_installation_configuration/configuring-aws-outposts.adoc

:_mod-docs-content-type: PROCEDURE
[id="aws-outposts-prereq-storage-csi_{context}"]
= Preparing the AWS EBS CSI driver to support AWS Outposts

//to-do: it would be better to employ reuse here so that these steps can't become outdated if something changes in the storage procedures. Need to make snippets from the following modules (in consultation with Storage writer):
//* modules/persistent-storage-csi-sc-managing-cli.adoc (change to Unmanaged)
//* modules/dynamic-provisioning-change-default-class.adoc (change default SC)

> You must update the default storage class for the AWS Elastic Block Store (EBS) Container Storage Interface (CSI) driver to be compatible with AWS Outposts.

Reviewer: Since we `Specifies a taint to prevent user workloads from being scheduled on nodes that have the `node-role.kubernetes.io/outposts` label.`,  from my understanding, change the default storageclass to `gp2-csi` become an optional step. If user doesn't change the default storageclass, deploy application with persistent volume on outposts nodes should specified the `gp2-csi` storageclass.
```suggestion
Better to update the default storage class for the AWS Elastic Block Store (EBS) Container Storage Interface (CSI) driver to be compatible with AWS Outposts.
```

Author: yes - I was asked to remove this topic in favor of the approach using taints. I had kept it in the PR for reference but i will pull the info that is still valid from it and remove it to prevent confusion.

// Module included in the following assemblies:
//
// * post_installation_configuration/configuring-aws-outposts.adoc

:_mod-docs-content-type: PROCEDURE
[id="aws-outposts-prereq-storage-csi_{context}"]
= Preparing the CSI driver to support AWS Outposts

> You must update the Container Storage Interface (CSI) driver default storage class to be compatible with AWS Outposts.

Reviewer 2: Which CSI driver are you refering to? AWS has Elastic Block Store (EBS) and Elastic File Service (EFS). Looks like EBS from what you say later in this doc, so you may want to call this out here.

> To prevent the CSI Operator from recreating the initial default storage class, you must set its state to *Unmanaged*.

Reviewer 2: To help out users, you can xref them to this topic for info about managing the default storage class and how to change this parameter: https://access.redhat.com/documentation/en-us/openshift_container_platform/4.14/html/storage/using-container-storage-interface-csi#overview.

Author: Per our conversation: leaving as-is for now, with plan for snippet-based reuse of content in the respective modules. Leaving a to-do note in this module with the next steps to do that.

Thank you again for looking this over and for your advice/expertise! :)

The driver requires the ARN to create volumes in an AWS Outposts instance.

{product-title} clusters on AWS include the `gp3-csi` and `gp2-csi` storage classes.
These classes correspond to Amazon Elastic Block Store (EBS) gp3 and gp2 volumes.
{product-title} clusters use the `gp3-csi` storage class by default, but AWS Outposts does not support EBS gp3 volumes.

.Prerequisites

* You have installed an {product-title} cluster into a custom VPC on AWS.

* You have access to the cluster using an account with `cluster-admin` permissions.

* You have installed the {oc-first}.

.Procedure

. Get the storage driver provisioner name by running the following command:
+
[source,terminal]
----
$ oc get storageclass
----
+
.Example output
[source,text]
----
NAME                PROVISIONER       RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
gp2-csi             ebs.csi.aws.com   Delete          WaitForFirstConsumer   true                   7h43m
gp3-csi (default)   ebs.csi.aws.com   Delete          WaitForFirstConsumer   true                   7h43m
----
+
In this example, the provisioner name is `ebs.csi.aws.com`.

> . Set the the `spec.storageClassState` field in the `ClusterCSIDriver` object to `Unmanaged` by running the following command:

Reviewer 2: Setting the SC to unmanaged is covered in the link I provided above.

+
[source,terminal]
----
$ oc patch clustercsidriver <provisioner_name> \
  --type=merge \
  -p "{\"spec\":{\"storageClassState\":\"Unmanaged\"}}"
----
+
where `<provisioner_name>` is the value of `PROVISIONER` in the `oc get storageclass` command output.

> . Set the default storage class to use gp2 volumes by running the following commands:

Reviewer 2: Changing the default SC is covered in this topic: https://access.redhat.com/documentation/en-us/openshift_container_platform/4.14/html/storage/using-container-storage-interface-csi#change-default-storage-class_persistent-storage-csi-sc-manage
